PocketKit/Tests/SyncTests/PocketSourceTests+savedItemEvents.swift

```diff
@@ -46,4 +46,26 @@ extension PocketSourceTests {
         XCTAssertEqual(notifications, [])
         XCTAssertFalse(space.context.hasChanges)
     }
+
+    func test_events_whenOSNotificationCenterPostsUnresolvedItemCreatedNotification_enqueuesASaveItemOperation() throws {
+        let operationStarted = expectation(description: "operationStarted")
+        operations.stubSaveItemOperation { _, _, _, _, _ in
+            return TestSyncOperation {
+                operationStarted.fulfill()
+            }
+        }
+
+        let source = subject()
+
+        let savedItem = try! space.seedSavedItem()
+        let unresolved: UnresolvedSavedItem = space.new()
+        unresolved.savedItem = savedItem
+        try space.save()
+
+        osNotificationCenter.post(name: .unresolvedSavedItemCreated)
+
+        wait(for: [operationStarted], timeout: 1)
+
+        try XCTAssertEqual(space.fetchUnresolvedSavedItems(), [])
```

Comment:
I feel as if the test name (`enqueuesASaveItemOperation`) implies more than just checking if we have no unresolved saved items. Can we dig into the persisted tasks to make sure we have a `saveItem` task ready?

Reply:
I see your point but line 67 above is actually the "assertion" that a saveItem task is created (and started). The operationFactory is stubbed out to provide a `TestSyncOperation` that fulfills the expectation when the operation is started.